skimage/util/compare.py

```diff
@@ -4,15 +4,15 @@
 from itertools import product
 
 
-def compare_images(image1, image2, method='diff', *, n_tiles=(8, 8)):
```

Comment:
In this case we could get away with making the arguments position-only, but I'm hesitant to really use that feature of Python. I feel like it's a lesser known aspect of Python and might confuse unnecessarily.

But I think, if we use a wrapper here that warns for a deprecation period, we can get away with eventually settling on `image0, image1`, cause regardless of how you used the old API and how long a user slept on the warning, the new API will always raise an error:
```python
compare_images(a, b)  # fine for old and new
compare_images(image1=a, image2=b)  # will raise
compare_images(a, image2=b) # will raise
```

Thinking about this, that might go for the `plot_matches` above as well, but renaming might still be easier.

Reply:
> In this case we could get away with making the arguments position-only, but I'm hesitant to really use that feature of Python. I feel like it's a lesser known aspect of Python and might confuse unnecessarily.

Right; I had to look it up to learn that all arguments before `/` in the function signature means they are positional-only. I don't think we have it anywhere in the codebase (as of now).